lms/views/basic_lti_launch.py

```diff
@@ -188,7 +194,64 @@ def url_configured_basic_lti_launch(self):
         LMS, which passes it back to us in each launch request. All we have to
         do is pass the URL to Via.
         """
-        return self.basic_lti_launch(self.request.parsed_params["url"])
+        groups = None
+
+        lti_user = self.request.lti_user
+        canvas = self.request.find_service(name="canvas_api_client")
+        if True:  # "is_group_assigment":
+            group_set_id = 121  # comming from assigment config?
+            course_id = self.request.params["custom_canvas_course_id"]
+            if lti_user.is_learner:
+                student_groups = canvas.get_course_groups(
+                    course_id,
+                    only_own_groups=True,
+                )
+                assignment_group = [
+                    g for g in student_groups if g["group_category_id"] == group_set_id
+                ]
+                if not assignment_group:
+                    print(
+                        "Student't doesn't belong to any of the groups on the assigment group set"
+                    )
+                    # handle this in some way
+                    raise ValueError("mmmm")
+                else:
+                    assignment_group = assignment_group[0]
+
+                groups = [
+                    HGroup.canvas_group(
+                        assignment_group["name"],
+                        assignment_group["id"],
+                        self.request.params["tool_consumer_instance_guid"],
+                        course_id,
+                    )
+                ]
+
+                self.request.find_service(name="lti_h").sync(
+                    groups,
+                    self.request.params,
+                )
+            elif lti_user.is_instructor:
```

Comment:
Very similar to the student part but here it gets all the groups in the group set.